shoop/admin/modules/orders/views/edit.py

```diff
@@ -99,32 +134,61 @@ def get_price_info(shop, customer, product, quantity):
     return pricing_mod.get_price_info(pricing_ctx, product, quantity=quantity)
 
 
-class OrderCreateView(TemplateView):
+class OrderEditView(CreateOrUpdateView):
     model = Order
     template_name = "shoop/admin/orders/create.jinja"
     context_object_name = "order"
     title = _("Create Order")
+    fields = []
 
     def get_context_data(self, **kwargs):
-        context = super(OrderCreateView, self).get_context_data(**kwargs)
+        context = super(OrderEditView, self).get_context_data(**kwargs)
         context["config"] = self.get_config()
         return context
 
+    def get_toolbar(self):
+        return Toolbar([])
+
     def get_config(self):
+        order = self.object
         shops = [encode_shop(shop) for shop in Shop.objects.filter(status=ShopStatus.ENABLED)]
         shipping_methods = ShippingMethod.objects.enabled()
         payment_methods = PaymentMethod.objects.enabled()
         return {
             "shops": shops,
             "countries": [{"id": code, "name": name} for code, name in list(countries)],
             "shippingMethods": [encode_method(sm) for sm in shipping_methods],
-            "paymentMethods": [encode_method(pm) for pm in payment_methods]
+            "paymentMethods": [encode_method(pm) for pm in payment_methods],
+            "orderId": order.pk,
+            "orderData": self.get_initial_order_data()
+        }
+
+    def get_initial_order_data(self):
+        order = self.object
+        if not order.pk:
+            return {}
+        return {
+            "shop": encode_shop(order.shop),
+            "lines": [
+                get_line_data_for_edit(order.shop, line) for line in order.lines.filter(
+                    type__in=[OrderLineType.PRODUCT, OrderLineType.OTHER]
```

Comment:
Discount lines is reprocessed/formed in confirm phase in order edit. Based on changes to original order the discount lines will change.